### ISSUE TITLE
[BETA] `ember s` exit silently without output

### ISSUE THREAD

**Issue by chancancode** [ember-cli/ember-cli#9216]
Reproduction here: https://github.com/ember-learn/super-rentals/commits/3-19-beta-failure

The tutorial build started failing a few days ago for this reason: https://github.com/ember-learn/super-rentals-tutorial/actions

Not sure if related, but both my local machine and the CI uses volta, so `ember s` goes through the volta shim, fwiw. But it should just be the same as running `node_modules/.bin/ember s` in the end. (And yes, I just tried, `node_modules/.bin/ember s` is the same.)

**Comment by rwjblue** [comment]
Hmm, this seems bad. Have you poked at it at all yet?

Anyone else seeing this with the betas?

**Comment by chancancode** [comment]
I haven't yet. By accident, I initially had port 4200 already taken, and it did fail with the "port already in use" error message, so I know _that_ works. (So it's running _something_.)

**Comment by chancancode** [comment]
I traced this down to [this loop](https://github.com/ember-cli/ember-cli/blob/c18f69d288f73efca566e6055c7d93d15e1c198c/lib/tasks/server/express-server.js#L103-L114).

The list of addons in this case (basically just the stock ones):

```
  'amd-transform',
  'tests-server-middleware',
  'history-support-middleware',
  'live-reload-middleware',
  'broccoli-watcher',
  'broccoli-serve-files',
  'proxy-server-middleware',
  'testem-url-rewriter',
  '@ember/optional-features',
  '@glimmer/component',
  'ember-auto-import',
  'ember-cli-babel',
  'ember-cli-app-version',
  'ember-cli-dependency-checker',
  'ember-cli-htmlbars',
  'ember-cli-uglify',
  'broccoli-asset-rev',
  'ember-cli-sri',
  'ember-data',
  'ember-export-application-global',
  'ember-fetch',
  'ember-load-initializers',
  'ember-maybe-import-regenerator',
  'ember-resolver',
  'ember-source',
  'ember-welcome-page',
  'loader.js',
  'ember-qunit',
  'qunit-dom'
```

I added `console.trace` at the beginning and end of the callback (with try/catch/finally). Consistently, that loop that `mapSeries` loop seems to successfully finish for the first four addons (`amd-transform`, `tests-server-middleware`, `history-support-middleware` and `live-reload-middleware`), but the process quits before it can start the fifth addon (`broccoli-watcher`).

**Comment by rwjblue** [comment]
Was resolved by #9240
